File: Linkedin/linkedin-become-a-programmer-foundations/3.programming-foundations-object-oriented-design-3/SpaceShip.py
# /**
#  * SpaceShip
#  */
class SpaceShip:

    # class variables
    toughness = 0

    #  instance variables
    def __init__(self):
        self.callSign = 'useless name'
        self.__shieldStrength = 100

    #  methods
    def fireMissile(self):
        return "Pew!"

    def reduceShield(self, amount):
        self.shieldStrength -= amount

    def getShieldStrength(self):
        return self.__shieldStrength


def main():
    myShip = SpaceShip()
    print(myShip.callSign)
    myShip.toughness += 1
    myShip.toughness += 1
    SpaceShip.toughness += 5
    SpaceShip.toughness += 1
    print(myShip.toughness)
    print(SpaceShip.toughness)


if __name__ == "__main__":
    main()

File: Linkedin/linkedin-become-a-programmer-foundations/3.programming-foundations-object-oriented-design-3/SpaceShip.java
/**
 * SpaceShip
 */
public class SpaceShip {

    // instance variables
    public String callSign;
    private int shieldStrength;

    // methods
    public String fireMissile() {
        return "Pew!";
    }

    public void reduceShield(int amount) {
        shieldStrength -= amount;
    }

    public int getShieldStrength() {
        return shieldStrength;
    }
}